code/game/objects/items/tools.dm

```diff
@@ -463,12 +463,30 @@
 		if(length(container.reagents.reagent_list) > 1)
 			to_chat(user, "<span class='warning'>[container] has too many chemicals mixed into it. You wouldn't want to put the wrong chemicals into [src].</span>")
 			return ..()
-		if(amountNeeded > 0 && container.reagents.has_reagent("welding_fuel"))
+		if(amountNeeded <= 0)
+			to_chat(user, "<span class='warning'>[src] is already full.</span>")
+			return ..()
+		if(container.reagents.has_reagent("welding_fuel"))

```

Comment:
if it has welding fuel, you only pass in the welding fuel?
Magically able to split apart the two chemicals by pure force of will and only pour in the welding fuel.

Reply:
Perhaps it's sensing via weight and plasma is lighter than welding fuel?

Reply:
No, because if you have both plasma and welding fuel in the beaker you'll never get to this point. You'd fail the check at line 463.

Reply:
That check is dumb.
People should be able to fuck up, it's part of the game.

Reply:
I guess that might make the return a bit redundant, though